reactor-core/src/main/java/reactor/util/repeat/RepeatSpec.java

```diff
@@ -0,0 +1,39 @@
+package reactor.util.repeat;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.util.Logger;
+import reactor.util.Loggers;
+import reactor.util.context.ContextView;
+
+import java.util.function.Predicate;
+
+public class RepeatSpec extends Repeat {
+    static final Logger log = Loggers.getLogger(RepeatSpec.class);
+    private final long repeats;
+
+    RepeatSpec(Predicate<? super ContextView> repeatPredicate, long repeats) {
+        this.repeats = repeats;
+    }
+
+    static Repeat create(Predicate<? super ContextView> predicate, long times) {
+        return new RepeatSpec(predicate, times);
+    }
+
+    @Override
+    public Flux<Long> generateCompanion(Flux<RepeatSignal> other) {
+        return Flux.deferContextual(cv ->
+                other
+                        .contextWrite(cv)
+                        .concatMap(retryWhenState -> {
+                            RepeatSignal copy = retryWhenState.copy();
+                            long iteration = copy.getRepeatsSoFar();
+                            if (iteration > repeats) {
```

Comment:
I don't think this is the most graceful way to do this, was trying to get some inspiration from the Retry one but the case is slightly different there